packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts

```diff
@@ -7,218 +7,230 @@
 
 import { assert, expect } from 'chai';
 import * as path from 'path';
-import * as vscode from 'vscode';
-
-describe('LWC Intellisense Test Suite', function() {
-  let lwcDir: string;
-  let lwcExtension: vscode.Extension<any>;
-  before(async() => {
-    lwcDir = path.join(
-      vscode.workspace.workspaceFolders![0].uri.fsPath,
-      'force-app',
-      'main',
-      'default',
-      'lwc'
-    );
+import {
+  commands,
+  CompletionItem,
+  CompletionItemKind,
+  CompletionList,
+  extensions,
+  Extension,
+  Position,
+  Selection,
+  TextDocument,
+  TextEditor,
+  Uri,
+  window,
+  workspace
+} from 'vscode';
+
+describe('LWC Intellisense Integration Tests', () => {
+
+  let lwcExtension: Extension<any>;
 
-    lwcExtension = vscode.extensions.getExtension(
+  before(async () => {
+    lwcExtension = extensions.getExtension(
       'salesforce.salesforcedx-vscode-lwc'
-    ) as vscode.Extension<any>;
+    ) as Extension<any>;
     await lwcExtension.activate();
   });
 
-  afterEach(async () => {
-    try {
-      await vscode.commands.executeCommand(
-        'workbench.action.closeActiveEditor'
-      );
-    } catch (e) {
-      throw e;
-    }
+  it('LWC Extension Activation', async () => {
+    expect(lwcExtension.isActive);
   });
 
-  it('lwc extension activation', async function() {
-    expect(lwcExtension.isActive);
-  }).timeout(10000);
-
-  /**
-   * Test that lwc markup intellisense includes standard lwc tags and custom lwc tags
-   */
-  it('LWC Markup Intellisense', async () => {
-    const docUri = vscode.Uri.file(path.join(lwcDir, 'hello', 'hello.html'));
-    const doc = await vscode.workspace.openTextDocument(docUri);
-    const editor = await vscode.window.showTextDocument(doc);
-
-    // We have to have some text or we'll just get generic completions
-    const text = '<c-';
-    const startPosition = new vscode.Position(5, 0);
-    const endPosition = new vscode.Position(
-      startPosition.line,
-      startPosition.character + text.length
-    );
-    const rangeReplace = new vscode.Range(startPosition, endPosition);
-    await editor.edit(editBuilder => {
-      editBuilder.replace(rangeReplace, text);
+  describe('LWC JS Intellisense Test Suite', function() {
+
+    // Time taken to execute the command to fetch actualcompletion list, varies with different environment and system.
+    // tslint:disable-next-line:no-invalid-this
+    this.timeout(10000);
+    let lwcDir: string;
+    let docUri: Uri;
+    let doc: TextDocument;
+    let editor: TextEditor;
+    let text: string;
+    let startPosition: Position;
+    let endPosition: Position;
+    let selection: Selection;
+
+    beforeEach(async () => {
+      lwcDir = path.join(
+        workspace.workspaceFolders![0].uri.fsPath,
+        'force-app',
+        'main',
+        'default',
+        'lwc'
+      );
+      docUri = Uri.file(path.join(lwcDir, 'hello', 'hello.js'));
```

Comment:
just observation that the above two lines don't ever change so they could have been assigned and been const's outside the `beforeEach`

Reply:
Yes! Thank you. Now, since we are dealing with JS and Html separately, `doc` and `docUri` can be treated as consts.